config: update password hashes after auth_type change (#9250)
tarantool/tarantool: changelogs/unreleased/gh-8967-creds-consider-auth-type.md

## feature/config

> * Now the password hash (and salt) will be regenerated for users managed
>   by config, when `security.auth_type` is different from the `auth_type`
>   for the user (gh-8967).

Reviewer: ```suggestion
* Now a password hash (and salt) will be regenerated for users managed
  in a configuration file if `security.auth_type` differs from a user's `auth_type`
  (gh-8967).
```

Author: Applied.
